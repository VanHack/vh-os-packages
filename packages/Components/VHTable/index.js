import React from 'react'

import VHInfoTitleDescription from  '../InfoTitleDescription';
import VHSelect from '../Input/Select'
import Ordinal from '../Ordinal';

import { Container, List, ListItem, ItemContent, Title } from './styles'

/**
 * 
 * mainSkills: [
    {
      current: items[1],
      order: 0,
      yearOfExperience: items[1]
    },
    {
      current: items[2],
      order: 1,
      yearOfExperience: items[1]
    }
  ], props 
 */

const VHTable = (props) => {
  return (
  <Container>
    <VHInfoTitleDescription
      title={props.title}
      description={props.subtitle}
      data=""
    />
    <List>
      {props.list.map(item => (
        <ListItem key={item.order}>
          {props.ordinal && <Ordinal value={(item.order+1)} />}
          <ItemContent>
            <Title>{item.current.label}</Title>
            <VHSelect
              data='selectExperience'
              className='yearsOfExperience'
              currentItem={props.items[0]}
              items={props.items}
              onEvent={e => {
                console.log(e)
              }}
            />
          </ItemContent>
        </ListItem>
      ))}
    </List>
  </Container>)
}

VHTable.displayName = "VHTable"

export default VHTable;