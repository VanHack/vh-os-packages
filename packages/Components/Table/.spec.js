import React from 'react';
import VHTable from '.';
import { render } from '../../../setupTests';

describe('[Component] - VHTable', () => {
  // helper function to render default component used in tests
  const setup = props => {
    const handleEvent = e => jest.fn();
    const utils = render(
      <VHTable
        data="main"
        title="simple title"
        size={(props && props.size) || 3}
        onEvent={handleEvent}
        {...props}
      />
    );

    return {
      ...utils,
      handleEvent
    };
  };

  it('renders component without crashing', () => {
    const { getByTestId } = setup();
    expect(getByTestId('vh-table')).toMatchSnapshot();
  });

  it('renders VHTable with the correct TITLE', () => {
    const { container } = setup();
    const element = container.querySelector(
      '[class*="vh-title-description"] > p'
    );

    expect(element.textContent).toBe('simple title');
  });

  it('renders VHTable with ORDINAL', () => {
    const { container } = setup({ ordinal: true });

    const elements = container.querySelectorAll('[data*="vh-ordinal"]');
    expect(elements[0].textContent).toEqual('1st');
    expect(elements[1].textContent).toEqual('2nd');
    expect(elements[2].textContent).toEqual('3rd');
  });

  it('renders VHTable with BADGE', () => {
    const list = [
      {
        current: [{ label: 'item01', value: '01' }],
        order: 0,
        yearOfExperience: [{ label: '1 year', value: '1' }],
        badge: 'code test 90%'
      }
    ];
    const { container} = setup({ list, size: 1 });
    const element = container.querySelector('[data="vh-badge"]');

    expect(element.textContent).toEqual('code test 90%')
  });

  it('renders VHTble with the correct NUM OF LINES', () => {
    const rows_count = 5;
    const { container } = setup({ size: rows_count });
    const elements = container.querySelectorAll('[data*="selectExperience"]');

    expect(elements.length).toBe(rows_count);
  });

  it('renders VHTable selects with first option item', () => {
    const myOptions = [
      { label: 'First value', value: '0' },
      { label: '3 years', value: '3' }
    ];
    const { container } = setup({ options: myOptions });
    const elements = container.querySelectorAll('[class*="singleValue"]');

    expect(elements[0].textContent).toBe('First value');
  });

  it('renders VHTable with correct data attribute for SELECT', () => {
    const { container } = setup();
    const elements = container.querySelectorAll('[data*="selectExperience"]');

    // check if data attr if correct (parentDataAttr-itemDataAttr-index)
    elements.forEach((item, index) => {
      expect(item.getAttribute('data')).toEqual(
        `main-selectExperience-${index}`
      );
    });
  });

  it('renders VHTable with correct data attribute for INPUT', () => {
    const { container } = setup();
    const elements = container.querySelectorAll('[data*="inputExperience"]');

    // check if data attr if correct (parentDataAttr-itemDataAttr-index)
    elements.forEach((item, index) => {
      expect(item.getAttribute('data')).toEqual(
        `main-inputExperience-${index}`
      );
    });
  });
});
