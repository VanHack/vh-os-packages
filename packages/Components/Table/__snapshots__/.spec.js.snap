// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] - VHTable renders component without crashing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  padding: 0;
  margin: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 * {
  margin: 0 0 0 0;
  display: inline;
}

.c5 * {
  margin: 0 0 0 0;
  display: block;
}

.c3 {
  padding-left: 5px;
}

.c2 {
  margin: 0;
  padding: 0;
  color: #4f4f4f;
  font-family: Roboto;
  font-size: 16px;
  font-weight: 'lighter';
  line-height: 23.1px;
}

.c4 {
  margin: 0;
  padding: 0;
  color: #e0e0e0;
  font-family: Roboto;
  font-size: 14px;
  font-weight: 'regular';
  line-height: 21.5px;
}

.c6 {
  margin: 0;
  padding: 0;
  color: #9f9f9f;
  font-family: Roboto;
  font-size: 14px;
  font-weight: 'regular';
  line-height: 21.5px;
}

.c7 {
  margin: 0;
  padding: 0;
  color: #4f4f4f;
  font-family: Roboto;
  font-size: 14px;
  font-weight: 'regular';
  line-height: 21.5px;
}

.c12 {
  position: relative;
  margin-bottom: 21px;
}

.c12 div[class*="IndicatorsContainer"] div:first-of-type,
.c12 div[class*="IndicatorsContainer"] span {
  display: none;
}

.c12 div[class*="multiValue"] {
  background: #fff;
  border: 1px solid #ccc;
  border-radius: 30px;
  padding: 0 5px;
}

.c12 div[class*="multiValue"] div:nth-child(2):hover {
  background: transparent;
}

.c11 {
  border: none;
  border-radius: 6px;
  color: #333333;
  font-size: 14px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.53;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  padding: 9px 24px;
  width: 100%;
  background: transparent;
  outline: 0;
  box-sizing: border-box;
}

.c11::-webkit-input-placeholder {
  font-size: 14px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.53;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #9f9f9f;
}

.c11::-moz-placeholder {
  font-size: 14px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.53;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #9f9f9f;
}

.c11:-ms-input-placeholder {
  font-size: 14px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.53;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #9f9f9f;
}

.c11::placeholder {
  font-size: 14px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.53;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #9f9f9f;
}

.c8 {
  list-style: none;
  padding: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 50px;
  border-bottom: 1px solid #ccc;
}

.c10 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 input[type="text"] {
  padding: 0;
  width: 200px;
  font-weight: bold;
}

.c10 [data*="selectExperience"] {
  width: 120px;
  padding: 0;
  margin: 0;
  border: 0;
}

.c10 [data*="selectExperience"] div[class*="control"] {
  border: 0;
}

.c10 [data*="selectExperience"] div[class*="control"]:hover {
  box-shadow: 0;
}

<div
  class=""
  data-testid="vh-table"
  size="3"
  title="simple title"
>
  <div
    class="c0"
  >
    <div
      class="c1 vh-title-description vh-info-title-description"
    >
      <p
        class="c2 vh-text-component "
        color="gray-90"
      >
        simple title
      </p>
      <div
        class="c3"
      >
        <p
          class="c4 vh-text-component "
          color="gray-30"
        />
      </div>
    </div>
    <div
      class="c5 vh-title-description vh-info-title-description"
    >
      <p
        class="c6 vh-text-component "
        color="gray-50"
      />
      <div
        class=""
      >
        <p
          class="c7 vh-text-component "
          color="gray-90"
        />
      </div>
    </div>
  </div>
  <ul
    class="c8"
  >
    <li
      class="c9"
    >
      <div
        class="c10"
      >
        <input
          class="c11 vh-input "
          data="main-inputExperience-0"
          placeholder="Type here"
          type="text"
          value=""
        />
        <div
          class="c12 yearsOfExperience"
          data="main-selectExperience-0"
          data-testid="vh-select"
          placeholder="Not Set"
        >
          <div
            class="yearsOfExperience css-2b097c-container"
          >
            <div
              class=" css-yk16xz-control"
            >
              <div
                class=" css-1hwfws3"
              >
                <div
                  class=" css-1wa3eu0-placeholder"
                  style="opacity: 1; transition: opacity 1ms;"
                >
                  Select...
                </div>
                <div
                  class="css-b8ldur-Input"
                >
                  <div
                    class=""
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-2-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; font-weight: bold; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-6q0nyr-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="c9"
    >
      <div
        class="c10"
      >
        <input
          class="c11 vh-input "
          data="main-inputExperience-1"
          placeholder="Type here"
          type="text"
          value=""
        />
        <div
          class="c12 yearsOfExperience"
          data="main-selectExperience-1"
          data-testid="vh-select"
          placeholder="Not Set"
        >
          <div
            class="yearsOfExperience css-2b097c-container"
          >
            <div
              class=" css-yk16xz-control"
            >
              <div
                class=" css-1hwfws3"
              >
                <div
                  class=" css-1wa3eu0-placeholder"
                  style="opacity: 1; transition: opacity 1ms;"
                >
                  Select...
                </div>
                <div
                  class="css-b8ldur-Input"
                >
                  <div
                    class=""
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-3-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; font-weight: bold; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-6q0nyr-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="c9"
    >
      <div
        class="c10"
      >
        <input
          class="c11 vh-input "
          data="main-inputExperience-2"
          placeholder="Type here"
          type="text"
          value=""
        />
        <div
          class="c12 yearsOfExperience"
          data="main-selectExperience-2"
          data-testid="vh-select"
          placeholder="Not Set"
        >
          <div
            class="yearsOfExperience css-2b097c-container"
          >
            <div
              class=" css-yk16xz-control"
            >
              <div
                class=" css-1hwfws3"
              >
                <div
                  class=" css-1wa3eu0-placeholder"
                  style="opacity: 1; transition: opacity 1ms;"
                >
                  Select...
                </div>
                <div
                  class="css-b8ldur-Input"
                >
                  <div
                    class=""
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-4-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; font-weight: bold; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-6q0nyr-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
